examples/tinywl-new/surfacewrapper.cpp

```diff
@@ -32,13 +32,58 @@ SurfaceWrapper::SurfaceWrapper(QmlEngine *qmlEngine, WToplevelSurface *shellSurf
     m_surfaceItem->setResizeMode(WSurfaceItem::ManualResize);
     m_surfaceItem->setShellSurface(shellSurface);
 
+    shellSurface->safeConnect(&WToplevelSurface::requestMinimize, this, &SurfaceWrapper::requestMinimize);
+    shellSurface->safeConnect(&WToplevelSurface::requestCancelMinimize, this, &SurfaceWrapper::requestCancelMinimize);
+    shellSurface->safeConnect(&WToplevelSurface::requestMaximize, this, &SurfaceWrapper::requestMaximize);
+    shellSurface->safeConnect(&WToplevelSurface::requestCancelMaximize, this, &SurfaceWrapper::requestCancelMaximize);
+    shellSurface->safeConnect(&WToplevelSurface::requestMove, this, &SurfaceWrapper::requestMove);
+    shellSurface->safeConnect(&WToplevelSurface::requestResize, this, [this](WSeat *, Qt::Edges edge, quint32) {
+        requestResize(edge);
+    });
+    shellSurface->safeConnect(&WToplevelSurface::requestFullscreen, this, &SurfaceWrapper::requestFullscreen);
+    shellSurface->safeConnect(&WToplevelSurface::requestCancelFullscreen, this, &SurfaceWrapper::requestCancelFullscreen);
+
     connect(m_surfaceItem, &WSurfaceItem::implicitWidthChanged, this, [this] {
         setImplicitWidth(m_surfaceItem->implicitWidth());
         if (m_titleBar)
             m_titleBar->setWidth(m_surfaceItem->width());
     });
     connect(m_surfaceItem, &WSurfaceItem::heightChanged, this, &SurfaceWrapper::updateImplicitHeight);
     setImplicitSize(m_surfaceItem->implicitWidth(), m_surfaceItem->implicitHeight());
+    connect(this, &SurfaceWrapper::surfaceStateChanged, m_shellSurface, [this] {
+        switch (m_surfaceState.value()) {
+        case State::Maximized:
+            m_shellSurface->setMaximize(true);
+            break;
+        case State::Minimized:
+            m_shellSurface->setMinimize(true);
+            break;
+        case State::Fullscreen:
+            m_shellSurface->setFullScreen(true);
+            break;
+        case State::Normal: [[fallthrough]];
+        case State::Tiling: [[fallthrough]];
+        default:
+            break;
+        }
+    });
+    connect(this, &SurfaceWrapper::previousSurfaceStateChanged, m_shellSurface, [this] {
```

Comment:
Why connect to signal, can do in setSurfaceState?

Reply:
It's `BINDABLE_PROPERTY`,  Does it mean that it can be changed outside of setSurfaceState?

Reply:
Don't care this case, maybe can make the bindableSurfaceState to readonly.